### PR TITLE
[GBM] implement missing OnDisplayLost/OnDisplayReset notifications

### PR DESCRIPTION
**Pull request by a1rwulf**
See title.

### CODE REVIEW

File: xbmc/windowing/gbm/WinSystemGbm.cpp [hunk 1]
```diff
@@ -246,3 +265,17 @@ void CWinSystemGbm::Unregister(IDispResource *resource)
     m_resources.erase(i);
   }
 }
+
+void CWinSystemGbm::OnLostDevice()
+{
+  CLog::Log(LOGDEBUG, "%s - notify display change event", __FUNCTION__);
+
+  // make sure renderer has no invalid references
+  KODI::MESSAGING::CApplicationMessenger::GetInstance().SendMsg(TMSG_RENDERER_FLUSH);
```
Comment (FernetMenta) [inline xbmc/windowing/gbm/WinSystemGbm.cpp]: this can be dropped. add m_renderManager.Flush to CVideoPlayer::OnLostDisplay

Comment (a1rwulf) [reply xbmc/windowing/gbm/WinSystemGbm.cpp]: I guess that means, I should remove it for all the windowing systems right?

Comment (FernetMenta) [reply xbmc/windowing/gbm/WinSystemGbm.cpp]: Up to you. Highly welcome but not needed in scope of this pr

File: xbmc/windowing/gbm/WinSystemGbm.cpp [hunk 2]
```diff
@@ -246,3 +265,17 @@ void CWinSystemGbm::Unregister(IDispResource *resource)
     m_resources.erase(i);
   }
 }
+
+void CWinSystemGbm::OnLostDevice()
```
Comment (lrusak) [inline xbmc/windowing/gbm/WinSystemGbm.cpp]: can this belong to CWinSystem and be used by other windowing systems? it doesn't seem specific to just this windowing system.

Comment (FernetMenta) [reply xbmc/windowing/gbm/WinSystemGbm.cpp]: it's not not called from the outside and implemention is specific to implementatios. hence makes not much sense to promote it up the hierarchy

File: xbmc/windowing/gbm/WinSystemGbm.cpp (hunk 1 above)
Comment (a1rwulf) [reply xbmc/windowing/gbm/WinSystemGbm.cpp]: I've used the FlushRenderer that's present in VideoPlayer already.
It does m_renderManager.Flush(true).
I dropped the AppMessenger for the other windowing methods as well.

Comment (a1rwulf) [reply xbmc/windowing/gbm/WinSystemGbm.cpp]: I've used the FlushRenderer that's present in VideoPlayer already.
It does m_renderManager.Flush(true).
I dropped the AppMessenger for the other windowing methods as well.

**Review by FernetMenta — APPROVED** [review]

**Comment by a1rwulf** [comment]
jenkins build this please

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 3]
```diff
@@ -4875,6 +4875,7 @@ void CVideoPlayer::OnLostDisplay()
   m_VideoPlayerVideo->SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_PAUSE, true), 1);
   m_clock.Pause(true);
   m_displayLost = true;
+  FlushRenderer();
```
Comment (boot2k3) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: This string ` FlushRenderer();`  causes video freeze on 4-8 seconds on amlogic devices if display changes refresh rate during video start.

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/VideoPlayer.cpp]: why don't you create a ticket for amlogic related issues?

**Pull request merged**